numpy/f2py/src/fortranobject.c

```diff
@@ -701,6 +706,15 @@ PyArrayObject* array_from_pyobj(const int type_num,
     }
 
     descr = PyArray_DescrFromType(type_num);
+    /* compatibility with NPY_CHAR */
+    if (type_num == NPY_STRING) {
+        descr = PyArray_DescrNew(descr);
```

Comment:
one could use `PyArray_DESCR_REPLACE` but that decrefs and PyArray_DescrFromType does not incref basic types.
I assume that is actually a bug we just get away with it because we likely leak descriptors all over the place.